src/driver/read_fixed.rs

```diff
@@ -0,0 +1,64 @@
+use crate::buf::fixed::FixedBuf;
+use crate::buf::{IoBuf, IoBufMut, Slice};
+use crate::driver::op::{self, Completable};
+use crate::driver::{Op, SharedFd};
+use crate::BufResult;
+
+use std::io;
+
+pub(crate) struct ReadFixed {
+    /// Holds a strong ref to the FD, preventing the file from being closed
+    /// while the operation is in-flight.
+    #[allow(dead_code)]
+    fd: SharedFd,
+
+    /// The in-flight buffer.
+    buf: Slice<FixedBuf>,
```

Comment:
This is very cool that the Slice of a FixedBuf can be used like this because the uring interface expressly allows the fixed read (and fixed write) to be done from addresses not necessarily at the start of the registered fixed buffer.